webapp/channels/src/components/activity_log_modal/activity_log_modal.tsx

```diff
@@ -47,110 +49,79 @@ export type Props = {
     };
 }
 
-type State = {
-    show: boolean;
-}
-
-export default class ActivityLogModal extends React.PureComponent<Props, State> {
-    static propTypes = {
-
-    };
-
-    constructor(props: Props) {
-        super(props);
-
-        this.state = {
-            show: true,
-        };
-    }
-
-    submitRevoke = (altId: string, e: React.MouseEvent) => {
+const ActivityLogModal = ({
+    currentUserId,
+    sessions,
+    locale,
+    onHide,
+    actions: {
+        getSessions,
+        revokeSession,
+    },
+}: Props) => {
+    const submitRevoke = useCallback((altId: string, e: React.MouseEvent) => {
         e.preventDefault();
         const modalContent = (e.target as Element)?.closest('.modal-content');
         modalContent?.classList.add('animation--highlight');
         setTimeout(() => {
             modalContent?.classList.remove('animation--highlight');
         }, 1500);
-        this.props.actions.revokeSession(this.props.currentUserId, altId).then(() => {
-            this.props.actions.getSessions(this.props.currentUserId);
+        revokeSession(currentUserId, altId).then(() => {
+            getSessions(currentUserId);
         });
-    };
-
-    onShow = () => {
-        this.props.actions.getSessions(this.props.currentUserId);
-    };
-
-    onHide = () => {
-        this.setState({show: false});
-    };
+    }, [currentUserId, revokeSession, getSessions]);
 
-    componentDidMount() {
-        this.onShow();
-    }
+    useEffect(() => {
+        getSessions(currentUserId);
+    }, [currentUserId, getSessions]);
```

Comment:
Nitpick: This is "changing the behavior". The linter probably told you to add all the dependencies, but the reality here is that the original behavior is "only get the sessions on mount". Of course, we don't expect currentUserId nor getSessions to ever change due to their nature, so this will behave the same as if `onMount`. 0/5 on whether to remove the dependencies or not.